### ISSUE TITLE
I ported inference to OPT

### ISSUE THREAD

**Issue by Ph0rk0z** [johnsmith0031/alpaca_lora_4bit#26]
https://github.com/Ph0rk0z/GPTQ-Merged/commit/24b57d83467ade90116869729668be7d9c7e4d69

I got it doing inference on opt. Not sure on training as I've not trained a single thing yet.

Inference for the 6b wasn't bad at least. Have to convert the 13b or the 30b but it takes time and will be obsoleted by the new GPTQ.

`Output generated in 2.25 seconds (5.77 tokens/s, 13 tokens, context 77)`

Model to test with : https://huggingface.co/autobots/opt-6b-4-bit/tree/main

opt-13b.. took 4 hours to quantize. 

```
Output generated in 4.03 seconds (2.98 tokens/s, 12 tokens, context 97)
Output generated in 8.71 seconds (0.92 tokens/s, 8 tokens, context 1848)
Output generated in 16.67 seconds (2.34 tokens/s, 39 tokens, context 1782)
```

I will upload it soon and attempt GPT-J and GPT-NeoX. 

I guess it would be nice to have a good way to check coherence on these.

**Comment by Ph0rk0z** [comment]
So today I have made it generic and now GPT-J works.. like pygmalion and all. https://github.com/Ph0rk0z/GPTQ-Merged/commit/ca722fddfb9fed3137d270dea6add0c5640484ea


But neox for some reason fails.

```
Loading settings from settings.json...
Loading GPT-NeoXT-Chat-Base-20B...
Found models/GPT-NeoXT-Chat-Base-20B-4bit.pt
Loading Auto Model ...
Traceback (most recent call last):
  File "/home/mint/text-generation-webui-testing/server.py", line 234, in <module>
    shared.model, shared.tokenizer = load_model(shared.model_name)
  File "/home/mint/text-generation-webui-testing/modules/models.py", line 101, in load_model
    model = load_quantized(model_name)
  File "/home/mint/text-generation-webui-testing/modules/GPTQ_loader.py", line 73, in load_quantized
    model, tokenizer = load_auto_model_4bit_low_ram(path_to_model, f"models/{pt_model}.pt" )
  File "/home/mint/text-generation-webui-testing/repositories/GPTQ-Merged/autograd_4bit.py", line 219, in load_auto_model_4bit_low_ram
    model = accelerate.load_checkpoint_and_dispatch(model=model, checkpoint=model_path, device_map='auto')
  File "/home/mint/miniconda3/envs/textgen/lib/python3.10/site-packages/accelerate/big_modeling.py", line 479, in load_checkpoint_and_dispatch
    load_checkpoint_in_model(
  File "/home/mint/miniconda3/envs/textgen/lib/python3.10/site-packages/accelerate/utils/modeling.py", line 862, in load_checkpoint_in_model
    set_module_tensor_to_device(model, param_name, param_device, value=param, dtype=dtype)
  File "/home/mint/miniconda3/envs/textgen/lib/python3.10/site-packages/accelerate/utils/modeling.py", line 131, in set_module_tensor_to_device
    raise ValueError(f"{module} does not have a parameter or a buffer named {tensor_name}.")
ValueError: Autograd4bitQuantLinear() does not have a parameter or a buffer named weight.

```
Maybe it does not have lm_head? 


https://github.com/Ph0rk0z/text-generation-webui-testing/commits/main and you can try it out without disturbing regular textgen ui or hard-coded patches.


`python server.py --gptq-bits 4 --model pygmalion-6b_dev --autograd --cai-chat --gptq-model-type gptj`


Next up get the multi-gpu working somehow and submit a PR back here.

**Comment by Ph0rk0z** [comment]
Probably can close this since it's working for multiple modles.
